crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -265,6 +505,130 @@ pub struct QuiltVersionV1;
 
 impl QuiltVersionV1 {
     const QUILT_VERSION_BYTE: u8 = 0x00;
+    const BLOB_HEADER_SIZE: usize = 7;
+    const EXTENSION_SIZE_BYTES_LENGTH: usize = 4;
+
+    /// Returns the total size of the encoded blob.
+    pub fn compute_encoded_blob_size(blob: &QuiltStoreBlob) -> Result<usize, QuiltError> {
+        let mut extension_size = 0;
+
+        let identifier_bytes_size = bcs::serialized_size(&blob.identifier)
+            .map_err(|e| QuiltError::Other(format!("Failed to compute identifier size: {}", e)))?;
+        let identifier_size = u16::try_from(identifier_bytes_size).map_err(|e| {
+            QuiltError::Other(format!(
+                "Identifier size exceeds maximum allowed value: {}",
+                e
+            ))
+        })?;
+
+        extension_size += identifier_size as usize + BLOB_IDENTIFIER_BYTES_LENGTH;
+
+        let total_size = extension_size + blob.data().len() + QuiltVersionV1::BLOB_HEADER_SIZE;
+        Ok(total_size)
+    }
```

Comment:
Nit: This feels overly complicated to me. How about the following?
```suggestion
    pub fn compute_encoded_blob_size(blob: &QuiltStoreBlob) -> Result<usize, QuiltError> {
            let identifier_size = bcs::serialized_size(&blob.identifier)
            .map_err(|e| QuiltError::Other(format!("Failed to compute identifier size: {}", e)))?;
        if identifier_size >= MAX_BLOB_IDENTIFIER_BYTES_LENGTH {
            return Err(QuiltError::Other(format!(
                "identifier size exceeds maximum allowed value: {}",
                MAX_BLOB_IDENTIFIER_BYTES_LENGTH
            )));
        }

        Ok(identifier_size
            + BLOB_IDENTIFIER_BYTES_LENGTH
            + blob.data().len()
            + QuiltVersionV1::BLOB_HEADER_SIZE)
```

This requires an additional constant:
```rust
/// The maximum number of bytes for the identifier of the blob.
const MAX_BLOB_IDENTIFIER_BYTES_LENGTH: usize = 1 << (8 * BLOB_IDENTIFIER_BYTES_LENGTH) - 1;
```